### ISSUE TITLE
Question about otpd evaluation

### ISSUE THREAD

**Issue by jeanwsr** [pyscf/pyscf-forge#94]
https://github.com/pyscf/pyscf-forge/blob/bb743d16ecf997d935e32a3740f41991ad57882a/pyscf/mcpdft/otpd.py#L39-L40
https://github.com/pyscf/pyscf-forge/blob/bb743d16ecf997d935e32a3740f41991ad57882a/pyscf/mcpdft/otfnal.py#L86-L87
https://github.com/pyscf/pyscf-forge/blob/bb743d16ecf997d935e32a3740f41991ad57882a/pyscf/mcpdft/otfnal.py#L139-L140

In function `get_ontop_pair_density`, it looks like the value of `deriv` is always 1 for translated GGA, according to the value of `dens_deriv` in `otfnal.energy_ot`. However, the Pi 1st derivs are not used for a translated GGA energy calculation. So I guess the `deriv` passed in here may be wrong?

**Comment by matthew-hennefarth** [comment]
I think you might be correct, and that we are computing the gradient of $\Pi$ unnecessarily for tGGA. One should probably be passing [`Pi_deriv`](https://github.com/pyscf/pyscf-forge/blob/bb743d16ecf997d935e32a3740f41991ad57882a/pyscf/mcpdft/otfnal.py#L125) to the `get_ontop_pair_density` instead of the `dens_deriv`.

**Comment by jeanwsr** [comment]
yeah, in `otfnal.energy_ot` we can simply set `Pi_deriv=0`, but there are a few more calls to this function in other parts (https://github.com/search?q=repo%3Apyscf%2Fpyscf-forge%20get_ontop_pair_density&type=code). I'm not sure how to correct them.
